contracts/contracts/governance/Governor.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/ITimelock.sol";
+
+// Modeled off of Compound's Governor Alpha
+contract Governor {
+
+    /// @notice The address of the  Timelock
+    ITimelock public timelock;
+
+    /// @notice The address of the Governor Guardian
+    address public guardian;
+
+    /// @notice The total number of proposals
+    uint public proposalCount;
+
+    struct Proposal {
+        /// @notice Unique id for looking up a proposal
+        uint id;
+
+        /// @notice Creator of the proposal
+        address proposer;
+
+        /// @notice The timestamp that the proposal will be available for execution, set once the vote succeeds
+        uint eta;
+
+        /// @notice the ordered list of target addresses for calls to be made
+        address[] targets;
+
+        /// @notice The ordered list of values (i.e. msg.value) to be passed to the calls to be made
+        uint[] values;
+
+        /// @notice The ordered list of function signatures to be called
+        string[] signatures;
+
+        /// @notice The ordered list of calldata to be passed to each call
+        bytes[] calldatas;
+
+        /// @notice Flag marking whether the proposal has been executed
+        bool executed;
+    }
+
+    /// @notice The official record of all proposals ever proposed
+    mapping (uint => Proposal) public proposals;
+
+    /// @notice An event emitted when a new proposal is created
+    event ProposalCreated(uint id, address proposer, address[] targets, uint[] values, string[] signatures, bytes[] calldatas, string description);
+
+    /// @notice An event emitted when a proposal has been queued in the Timelock
+    event ProposalQueued(uint id, uint eta);
+
+    /// @notice An event emitted when a proposal has been executed in the Timelock
+    event ProposalExecuted(uint id);
+    
+    uint256 public constant MAX_OPERATIONS = 16;
+
+    /// @notice Possible states that a proposal may be in
+    enum ProposalState {
+        Queued,
+        Expired,
+        Executed
+    }
+
+
+    constructor(address timelock_, address guardian_) public {
+        timelock = ITimelock(timelock_);
+        guardian = guardian_;
+    }
+
+    function proposeAndQueue(address[] memory targets, uint[] memory values, string[] memory signatures, bytes[] memory calldatas, string memory description) public returns (uint) {
+        require(msg.sender == guardian, "Governor::proposeAndQueue: sender must be gov guardian");
+        require(targets.length == values.length && targets.length == signatures.length && targets.length == calldatas.length, "Governor::propose: proposal function information arity mismatch");
```

Comment:
lol. I had to lookup what "arity" means. fancy wording cPan!

Reply:
not mine this is mostly compound